app/messages/app_manifest_message.rb

```diff
@@ -128,7 +127,7 @@ def app_lifecycle_hash
                          cnb_lifecycle_data
                        elsif requested?(:docker)
                          docker_lifecycle_data
-                       else
+                       elsif requested?(:buildpacks) || requested?(:buildpack) || requested?(:stack)
```

Comment:
I've been debugging an issue where if you push a cnb app where the lifecycle is not specified in the manifest, but a buildpack is, then you get the following error:

```
cf push node-cnb --buildpack node
For application 'node-cnb': Lifecycle type cannot be changed from cnb to buildpack
FAILED
```

I believe it's because of this line, since specifying a buildpack doesnt necessarily mean the app is using the `buildpack` lifecycle.  The same would be true for stack I believe.  Not sure how to resolve this yet, I will keep looking into it tomorrow.